xarray/backends/api.py

```diff
@@ -1400,6 +1408,9 @@ def to_netcdf(
             raise ValueError(
                 f"unrecognized option 'invalid_netcdf' for engine {engine}"
             )
+    if auto_complex is not None:
+        kwargs["auto_complex"] = auto_complex
+
```

Comment:
Continuing the discussion above about making `auto_complex: bool`, then this could be:

```suggestion
    if auto_complex:
        if engine != "netcdf4":
            raise ValueError(
                f"unrecognised option 'auto_complex' for engine {engine}"
            )
        
        kwargs["auto_complex"] = auto_complex

```

This would mean we couldn't default to having this on though.

Reply:
I think the current approach is the minimal intrusive one. We might switch the default value when netcdf4-python 1.7.0 is the minimum required dependency.

Reply:
Yes, that's reasonable